analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -133,9 +134,16 @@ class Analyzer(private val config: AnalyzerConfiguration, private val labels: Ma
                 async {
                     val results = manager.resolveDependencies(files, labels)
 
-                    // By convention, project ids must be of the type of the respective package manager.
+                    // By convention, project ids must be of the type of the respective package manager. An exception
```

Comment:
Would it make sense to just map the type inside the flutter implementation from "Gradle" to "Flutter", so this special case can be avoided?

Reply:
This would hide some information what the actual package manager of the project is.
However, that might be ok, because it doesn't really make sense to scan a Gradle / CocoaPods project managed by Flutter by their own.
But in extension would this also mean to map the `Pub` and the `CococaPods` project to the type `Flutter`?

All these projects would end up in the dependency tree:
```
Flutter:gradle-subproject:bbb
Flutter:pub-project:aaa
    dependencies
        Flutter:gradle-subproject:bbb
```

IMO it's better to have this special case in this check, than to change the result.
What do you think?

Reply:
Are the `Pup` and `CocoaPods` types used for `projects` or just for `packages` / dependencies?

Reply:
At least `Pub` is also used for projects. See screenshot in the PR description.

Reply:
Oh, thanks for pointing to the screenshot. I missed that. I believe I'm fine as-is with the returned identifier types.

I'm lacking the flutter development knowledge in this question, maybe you can help: Is the parent child relation as-is now now in the right direction, or should the flutter project be a child project of the Gradle android app project?

For the use case "make a review for just the Android App (not the iOS one)" is it possible (as-is) to setup path excludes such that 
CocoaPods project is excluded, but not the shared Flutter one nor the Android one?

Reply:
Oh, thanks for pointing to the screenshot. I missed that. I believe I'm fine as-is with the returned identifier types.

I'm lacking the flutter development knowledge in this question, maybe you can help: Is the parent child relation as-is now now in the right direction, or should the flutter project be a child project of the Gradle android app project?

For the use case "make a review for just the Android App (not the iOS one)" is it possible (as-is) to setup path excludes such that CocoaPods project is excluded, but not the shared Flutter one nor the Android one?

Reply:
> I'm lacking the flutter development knowledge in this question, maybe you can help: Is the parent child relation as-is now now in the right direction, or should the flutter project be a child project of the Gradle android app project?

Yes, at least to my understanding that's correct. The final Android app is a combination of the code and dependencies from the Flutter project and the Android project.

> For the use case "make a review for just the Android App (not the iOS one)" is it possible (as-is) to setup path excludes such that CocoaPods project is excluded, but not the shared Flutter one nor the Android one?

The definition file path of the Gradle and CocoaPods projects is still relative to the analyzer root, so all path excludes should work as expected. However, as both are added to the same scope of the Pub project this could be an issue. Because if I'm right to completely exclude a project in a multi-module setup it's not enough to exclude the project with a path exclude, but also all dependencies on that project need to be excluded?

One option is to add the Gradle and CocoaPods projects not to the "dependencies" scope of the Pub project but to separate scopes, e.g. "android" and "ios" or "gradle" and "cocoapods", then it would be possible to use a combination of path and scope excludes. But maybe in a separate PR.

Reply:
Thanks @mnonnenmacher . I'd be ok with a separate PR. But I think that should be mandatory to do this as follow up to not loose the possibility to make separate reviews for iOS and Android releases. Does it make sense?

Reply:
@MarcelBochtler What's your preference?

Reply:
I'll change the scope of the child Gradle project to `android` in this PR.

Reply:
@fviernau, @mnonnenmacher The Gradle project is now added to the `android` scope.
I also updated the screenshot in https://github.com/oss-review-toolkit/ort/pull/4831#issue-798872331 to reflect the changes.
Can you please have a look if that is what you expected?

Hint: For Cocoapods / iOS, this will be done in a separate PR, as the `Pub` implementation currently doesn't support CocoaPods at all.

Reply:
Thanks @MarcelBochtler !